docs/cpp_docs.rst

```diff
@@ -26,94 +45,540 @@ The beginning of both header and source code files should include
+
+You must redocument the parameters if the template specialization redefines any parameters (e.g., if the generic parameter ``T`` becomes ``T*`` in the specialization) or if it places additional restrictions on their values.
+
+.. _cpp-doxygen-parameters:
+
+Function/Method Parameters
+--------------------------
+
+A series of ``@param`` tags, usually one for each parameter. Each tag should have a description following the parameter name. You do *not* usually need to document default arguments; Doxygen will provide the default automatically. If the description extends over multiple lines, each line after the first must be indented.
+
+Parameters should be listed in the same order as they appear in the function or method signature.
+
+``@param`` should be given with the ``[in]``, ``[out]``, or ``[in,out]`` tag if the function method contains any output parameters. The ``[in]`` tag is optional if all parameters are input, even if other functions or methods in the same class or package use output parameters.
+
+.. code-block:: cpp
+
+   /**
+    * Compute mean and standard deviation for a collection of data.
+    *
+    * @param[out] mean the mean of `data`, or `NaN` if `data` is empty
+    * @param[out] stdDev the unbiased (sample) standard deviation, or `NaN`
+    *     if `data` contains fewer than 2 elements
+    * @param[in] data the data to analyze
+    */
+   std::ostream operator<<(double & mean, double & stdDev, std::vector<double> const & data);
+
+When two or more consecutive parameters have *exactly* the same description, they can be combined:
+
+.. code-block:: cpp
+
+   /**
+    * @param x,y the coordinates where the function is evaluated
+    */
+
+Do not use a space between the parameters, or Doxygen will not parse the command correctly.
+
+.. _cpp-doxygen-parameters-inline:
+
+Annotating Parameters with Inline Comments (historical)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+An alternative to the ``@param`` tag is to use an inline comment after each parameter, one per line.
+These comments are prefixed with ``///<``.
+
+This style is permitted for historical reasons, but should not be used in new code.
+If the parameter descriptions are too long to fit in a single line of source, the ``@param`` documentation method *must* be used.
+
+.. _cpp-doxygen-returns:
+
+Returns
+-------
+
+A ``@returns`` tag, followed by a description similar to the one for :ref:`cpp-doxygen-parameters`. If the returned value is a map, ensure that the key-value pairs are documented in the description.
+
+For consistency with Python documentation, always use ``@returns`` and not the synonymous ``@return``.
+
+.. _cpp-doxygen-throws:
+
+Throws
+------
+
+A series of ``@throws`` tags, one for each type of exception (see :ref:`the style guide <style-guide-cpp-5-36>`). Each tag should have a description following the exception type. If the description extends over multiple lines, each line after the first must be indented.
```

Comment:
Python's "Raises" section says "This section should be used judiciously—only for errors that are non-obvious or have a large chance of getting raised."  Something like that might be appropriate here; trying to get complete and exact exception documentation seems like it could easily become unbearable for both the writer and reader.